### ISSUE TITLE
[torch.export] Automate export constrains like in onnx dynamo

### ISSUE THREAD

**Issue by bhack** [pytorch/pytorch#136210]
### 🚀 The feature, motivation and pitch

We need to simplify the end user/dev life exporting with `torch.export` using an optional flag to automate the use of `torch.export.dynamic_shapes.refine_dynamic_shapes_from_suggested_fixes`.

This is already done as a fallback by the onnx/dynamo export at 
https://github.com/pytorch/pytorch/blob/e3aa5e2f6410727efead4970a4ec30569cf76881/torch/onnx/_internal/exporter/_capture_strategies.py#L127-L137

cc @ezyang @chauhang @penguinwu @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

**Comment by pianpwk** [comment]
Hi @bhack, we built the suggested fixes parser as a temporary solution for automatic dynamic shapes, but we've recently added `Dim.AUTO` for one-shot export - just specify `Dim.AUTO` for any dimension that's allowed to be dynamic, and export will figure out the min/max range and relations (and specialize to a constant if the program says so). I mentioned this in #136216 - let me know if it's for a separate issue.

**Comment by avikchaudhuri** [comment]
Hope the suggestion of using Dim.AUTO works, closing. Feel free to reopen in case it's not sufficient.
